{PROJECT_NAME}/Sources/Data/NetworkAPI/Core/NetworkAPIProtocol.swift

```diff
@@ -12,33 +11,12 @@ protocol NetworkAPIProtocol {
 }
 
 extension NetworkAPIProtocol {
-
     func request<T: Decodable>(
-        session: Session,
-        configuration: RequestConfiguration,
-        decoder: JSONDecoder
+        provider: MoyaProvider<RequestConfiguration>,
+        configuration: RequestConfiguration
     ) -> Single<T> {
-        return session.rx.request(
-            configuration.method,
-            configuration.url,
-            parameters: configuration.parameters,
-            encoding: configuration.encoding,
-            headers: configuration.headers,
-            interceptor: configuration.interceptor
-        )
-        .responseData()
-        .flatMap { _, data -> Observable<T> in
-            Observable.create { observer in
-                do {
-                    let decodable = try decoder.decode(T.self, from: data)
-                    observer.on(.next(decodable))
-                } catch {
-                    observer.on(.error(error))
-                }
-                observer.on(.completed)
-                return Disposables.create()
-            }
-        }
-        .asSingle()
+        provider.rx.request(configuration)
+            .filterSuccessfulStatusAndRedirectCodes()
```

Comment:
Correct me if I'm wrong. I don't think that we need to filter the status code here. If there's a server error with code 5xx, will this request have no response? 

`filterSuccessfulStatusAndRedirectCodes() filters status codes that are in the 200-300 range.`